### ISSUE TITLE
Can not change value of XMM registers during a diversion

### ISSUE THREAD

**Issue by aarzilli** [rr-debugger/rr#3208]
Sending the 'P' packet with an XMM register during a diversion returns 'OK' but the value of the register is actually never changed, logs:

```
[GdbConnection] raw request P46=00000000000000000000000000000000
[WARN write_register_arch()] Unhandled register name 70
[GdbConnection] write_flush: '$OK#9a'
```

**Comment by khuey** [comment]
I think that's actually DREG_64_YMM10H, but yes, setting it is not implemented.

**Comment by aarzilli** [comment]
Possible, I didn't check. I meant it loosely

**Comment by khuey** [comment]
If you apply

```diff
diff --git a/src/GdbServer.cc b/src/GdbServer.cc
index 00cb59b9..4d4cb146 100644
--- a/src/GdbServer.cc
+++ b/src/GdbServer.cc
@@ -634,6 +634,10 @@ void GdbServer::dispatch_debugger_request(Session& session,
         return;
       }
       if (req.reg().defined) {
+        if (req.reg().name == 70) {
+          dbg->reply_set_reg(false);
+          return;
+        }
         Registers regs = target->regs();
         regs.write_register(req.reg().name, req.reg().value, req.reg().size);
         target->set_regs(regs);
```

to rr, and then retry whatever gdb commands led you to this, what happens?

**Comment by aarzilli** [comment]
> to rr, and then retry whatever gdb commands led you to this, what happens?

I didn't encounter this through gdb, I actually found it while maintaining Delve's rr backend. Currently it can't call functions that take some arguments in floating point registers, due to a bug in delve. If I fix that bug then the functions do get called but, of course, they don't actually get the right arguments.
If it were to return an error it would fail with that error. I see that passing false to reply_set_reg actually results in an empty response packet, that might be a minor problem, since it means "unsupported request", rather than being an error (I did see the comment in reply_set_reg, I understand why it's this way).
